### DISCUSSION TITLE
Multi way switch

### DISCUSSION THREAD

**Discussion by farmio** [home-assistant/architecture#565] [Entity Models] [answered]
Hello 👋!

It would be handy for some integrations if they could provide multi way switch entities directly. So a switch / toggle / dropdown that can have one state out of several options defined by the integration.

Some example states would be
- Off, Manual, Automatic (e.g., pumps)
- Off, Low, Medium, High
- Auto, Locked ON, Locked OFF (e.g., motion detectors)

This could look something like fan speed modes or input_select.

<hr> 

I know this can be done with multiple 2-state switches or input_select using automations and services, but this seems error prone and not very user-friendly.

There are several community forum threads about this exact topic - some here for reference:
https://community.home-assistant.io/t/switch-with-several-states/47031
https://community.home-assistant.io/t/custom-multiple-state-switch/237
https://community.home-assistant.io/t/three-way-ui-switch-on-off-auto/137699

**Comment by frenck** [comment]
@thomasloven has a tri-state thingy I believe

**Reply by thomasloven** [reply]
Only as a POC, but I felt it didn't provide a good user experience so I shelved it for the time being.

Edit: There's a screenshot of it in OPs third link.

**Reply by frenck** [reply]
> but I felt it didn't provide a good user experience

Because?
Was that the state capability in general or simply the UI experience it provided?
Would that be a concern for the suggested above?

**Reply by thomasloven** [reply]
Mostly because it was not obvious what would happen when you clicked it.
Say a lamp was in `auto` mode. Half of the time you'd need to tap the button twice to toggle the lamp.
And that's only on/off - not even considering brightness or color.
But it was also a bit hard to wrap your head around what it actually did.

My idea was that a special tri-state entity would be given an "output entity", and generate a clone of that as an "input entity".
The input entity would be controlled by automations and stuff, and if the tristate was in auto mode, that would reflect on the output.

Then I realized this didn't need to be a special entity, but could be an input_select, and the rest could be easily handled in automations. So it didn't really add much value except as a frontend thing.

**Comment by emontnemery** [comment]
I think this makes sense.

We already have the `number` base entity, this would be the same but allow calls to set as well as state updates from a list of options instead of from a numerical range.

**Comment by farmio** [comment, marked as answer]
I think the new select entity solved this pretty neat. 
https://github.com/home-assistant/core/pull/51849
